### PR TITLE
GetUserUseCase added

### LINKED ISSUE #45: Adding a use case in user domain
## Description ##

The goal of this issue is to add a GetUsersUseCase to the user feature and to update its presentation accordingly so when running the app we can access those functionality
## Implementation ##
- Creating the use case 
- Creating the test for the use case
- Adding the use case to the presentation

## Definition of done ##
- [x] The options are visualized in the user menu
- [x] The use case pass the tests

### PR DESCRIPTION
**Pull request by alessandropaucare**
## Task description

To add another use case, one where we get all the users and to update the presentation to acces all the use cases and update the menu to access the respective methods

## How has it been implemented?

Adding another use case as usual following our architecutre and repository pattern

## Keywords

UseCase. Repository. Clean Architecture.

## Screenshots or Video

image
image

### CODE REVIEW

File: src/main/java/com/iesam/digLibrary/features/user/presentation/UserPresentation.java [hunk 1]
```diff
@@ -59,7 +59,33 @@ public static void updateUser() {
...
+            System.out.println("User does not exist");
+        }
+    }
+
 
+    public static void getAllUsers(){
+        GetUsersUseCase useCase =  new GetUsersUseCase(new UserDataRepository(new UserFileLocalDataSource()));
+        List<User> userList = useCase.execute();
+        if(!userList.isEmpty()) {
+            System.out.println("Next there is a list with all the users stored: ");
```
Comment (Fouad-Trabajo) [inline src/main/java/com/iesam/digLibrary/features/user/presentation/UserPresentation.java:83]: Igual este mensaje podría ponerlo en el presentation

File: src/main/java/com/iesam/digLibrary/features/user/domain/GetUserByIdUseCase.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.iesam.digLibrary.features.user.domain;
 
 public class GetUserByIdUseCase {
-    private UserRepository repository;
+    public final UserRepository repository;
```
Comment (Fouad-Trabajo) [inline src/main/java/com/iesam/digLibrary/features/user/domain/GetUserByIdUseCase.java]: Creo que los atributos del repositorio debería ser privados

**Review by Fouad-Trabajo — DISMISSED** [review]
Good cabron

File: src/main/java/com/iesam/digLibrary/features/user/domain/GetUserByIdUseCase.java (hunk 2 above)
Comment (alessandropaucare) [reply src/main/java/com/iesam/digLibrary/features/user/domain/GetUserByIdUseCase.java]: Deberia revisarlo.

**Review by Fouad-Trabajo — APPROVED** [review]

**Pull request merged**